pkg/util/flags/flags.go

```diff
@@ -82,24 +82,34 @@ func createRequiredFlagError(name string, flags []string) error {
 	return fmt.Errorf("%s are required for %s", strings.Join(flags, ","), name)
 }
 
-func SetLoggingFlags(flags *pflag.FlagSet) {
+func SetLoggingFlags(flags *pflag.FlagSet) error {
 	//Add commandline flags to the flagset. We will always write to stderr
 	//and not to a file by default
 	flags.AddGoFlagSet(flag.CommandLine)
-	flags.Set("logtostderr", "true")
-	flags.Parse(nil)
+	if err := flags.Set("logtostderr", "true"); err != nil {
+		return fmt.Errorf("failed to set flag %q. %v", "logtostderr", err)
+	}
+	if err := flags.Parse(nil); err != nil {
+		return fmt.Errorf("failed to parse logging flag. %v", err)
+	}
+	return nil
 }
 
 func SetFlagsFromEnv(flags *pflag.FlagSet, prefix string) error {
+	var allErrors error
 	flags.VisitAll(func(f *pflag.Flag) {
 		envVar := prefix + "_" + strings.Replace(strings.ToUpper(f.Name), "-", "_", -1)
 		value := os.Getenv(envVar)
 		if value != "" {
 			// Set the environment variable. Will override default values, but be overridden by command line parameters.
-			flags.Set(f.Name, value)
+			if err := flags.Set(f.Name, value); err != nil {
+				allErrors = fmt.Errorf("%v. %v", allErrors, err)
```

Comment:
You're just capturing a single error here. How about if we just return when we hit the first error?

Reply:
Made that change with respect to this comment.
https://github.com/rook/rook/pull/4498#discussion_r364877729

Reply:
Frm the `flags.VisitAll()` function we cannot return an error. Even if we want to return the error, it looks like a vendor change to me. @travisn